main.c

```diff
@@ -28,12 +31,12 @@
 #endif
 
 // additional 0 terminator is added by C
-static const char * const pkprefix = "== ed25519v1-public: type0 ==\0\0";
-#define pkprefixlen (29 + 3)
-static const char * const skprefix = "== ed25519v1-secret: type0 ==\0\0";
-#define skprefixlen (29 + 3)
-static const char * const checksumstr = ".onion checksum";
-#define checksumstrlen 15
+static const char pkprefix[] = "== ed25519v1-public: type0 ==\0\0";
+#define pkprefixlen (sizeof(pkprefix)) // three null-terminators included
+static const char skprefix[] = "== ed25519v1-secret: type0 ==\0\0";
+#define skprefixlen (sizeof(skprefix)) // three null-terminators included
+static const char checksumstr[] = ".onion checksum";
+#define checksumstrlen (sizeof(checksumstr) - NULLTERM_LEN)
```

Comment:
`NULLTERM_LEN` kinda makes it less clear that it's just 1. not sure if substitution is needed there

Reply:
I use such techniques to make code more self describing. This way intention of this subtraction is clear.

By the way, I don't really sure if C language have such convention, but it looks like using word `size` is good for size of object in memory, and `length` for something about data which is represented by the object. This thoughts are derived from semantics of `sizeof` and `strlen`.
So it was tempting to rename `*prefixlen` to `*prefixsize`  :), to hint the fact of binary nature of data described by this constants.

Maybe this is completely wrong in actual C code world, then ok. :)

Reply:
prefixlen->prefixsize is actually good idea, haven't though of that, but not something to be done on feature pull request. changing too much stuff in one commit makes it harder to review.

Reply:
about `NULLTERM_LEN`, it just seemed overkill to define it for `1`. and without understanding how C does null-termination, one wouldn't really get what `NULLTERM_LEN` does either so not sure if it's really helpful..

Reply:
> changing too much stuff in one commit makes it harder to review.

Oh, I squash it to one especially for PR, not sure why. It is all here:
https://github.com/heios/mkp224o/commits/write-keys-to-file
Just in case.

Reply:
> one wouldn't really get what NULLTERM_LEN does either

I came from C++ world, there I extensively use construction `sizeof('\0')` or `sizeof('\n')`, but char literals in C is actually `int` type, so this is wrong here. But perfectly describing what magic `1` is about, what are we fighting with in arbitrary complex contexts.

Anyway, in case of this PR I agree that it's better to jump back to magic constants.

Reply:
I would personally never do `sizeof('\0')` or such in C++; if not `1`, I would use `sizeof(char)`.
just like I never do `sizeof(10)` and would do `sizeof(int)` instead.